### ISSUE TITLE
PoolRegistration is broken with hw-cli 1.6.0

### ISSUE THREAD

**Issue by gitmachtl** [vacuumlabs/cardano-hw-cli#83]
```
Cardano-HW-CLI: 1.6.0
Cardano-CLI: 1.27.0
Cardano-Ledger-App: 2.3.2
```
_______________

Trying to do a PoolRegistration or a PoolUpdate with Cardano-HW-CLI **1.6.0** results in the following error:

**ERROR:** `invalid pool reward account type for legacy Ledger version`

The used Tx-RawBody:

```json
{
    "type": "TxBodyMary",
    "description": "",
    "cborHex": "83a50081825820d957e263f0b22febd67e1f61608a94d2b70f831f9c5e9965e4878de64967129c01018182581d6052e63f22c5107ed776b70f7b92248b02552fd08f3e747bc7450994411b000000012698416e021a0002f4a9031a01b5075404818a03581c7d0b94c37548cd44bf7
afc9e27068c73e86bf9137787b04e9032d9275820f658ba2f2b6539f9d7d79b1ab7dbec157e3a26b4a393563dfa532cb59c0e4e191b000000174876e8001b00000002540be400d81e82010a581de003d205532089ad2f7816892e2ef42849b7b52788e41b3fd43a6e01cf81581c03d205532089ad2f78
16892e2ef42849b7b52788e41b3fd43a6e01cf818400190bb9447f000001f682782c68747470733a2f2f6d792d69702e61742f746573742f74657374706f6f6c2e6d657461646174612e6a736f6e58207cd083e6682892afc2b4e315fca93784f43dccde8bae1daa81db0b587e27155f9ffff6"
}
```

The error occurs mid during the witness signing process.

Cardano-HW-CLI **1.5.0** is working normally

@DavidTranDucVL , maybe its better to take 1.6.0 down again until cardano-ledger-app 2.4.0 is merged?

**Comment by DavidTranDucVL** [comment]
Good point, for now marked as pre-release.

**Comment by spklpool** [comment]
This looks like the same error I reported in the ledgerjs-cardano-shelley repo 
https://github.com/vacuumlabs/ledgerjs-cardano-shelley/issues/110
I found an assert with the exact message in that repo so I assume that would be the appropriate place to report it.  
Am I understanding correctly that this is due to code released in cardano-hw-cli that needs the cardano-ledger-app 2.4.0 before it can work?

**Comment by DavidTranDucVL** [comment]
Newest version of cardano-hw-cli is developed with Cardano ledger app 2.4.0. As it isn't officially released yet, cardano-hw-cli version 1.6.0 is marked as prerelease.

> Am I understanding correctly that this is due to code released in cardano-hw-cli that needs the cardano-ledger-app 2.4.0 before it can work?

Yes

**Comment by gitmachtl** [comment]
cardano-ledger-app 2.4.1 is now officially available for the users, and it is working again with hw-cli 1.6.x. so, i will close this issue.
